_pages/vectorautodiff.md

```diff
@@ -0,0 +1,184 @@
+---
+title: "Vectorized Automatic Differentiation"
+layout: gridlay
+excerpt: ""
+sitemap: false
+permalink: /vectorautodiff/
+---
+
+## Introduction
+
+Vectorized Forward Mode Automatic Differentiation is a computational technique
+that combines two powerful concepts: vectorization and forward mode automatic
+differentiation. This approach is used to efficiently compute derivatives of
+functions with respect to multiple input variables by taking advantage of both
+parallel processing capabilities and the structure of the computation graph.
+
+## Need for Vectorized Automatic Differentiation
+
+In forward mode AD, the restriction is that the function can be only be 
+differentiated with respect to a single input variable. However, in many cases,
+it is desirable to differentiate a function with respect to multiple input
+variables. One way to do this is to use a vectorized version of forward mode AD.
+
+Without vector mode, for computing derivative of a function with n-dimensional
+input - forward mode requires n forward passes, i.e. one for each input variable.
+In vector mode, all these computations are batched together and computed in a
+single forward pass and the function is differentiated with respect to multiple
+input variables. This can help in reducing the overhead of computing an expensive
+operation in multiple forward passes. This can also help in utilizing the
+vectorization capabilities of the hardware.
+
+The output of the function is a vector of partial derivatives with respect to
+each input variable. Currently, Clad only supports vectorized version of forward
+mode AD.
+
+## Usage
+
+The following code snippet shows how one can request Clad to use vector mode for
+differentiation:
+
+```cpp
+#include "clad/Differentiator/Differentiator.h"
+
+double prod(double x, double y, double z) { return x*y*z; }
+
+int main(){
+    auto grad = clad::differentiate<clad::opts::vector_mode>(prod, "x,y");
+    double x = 3.0, y = 4.0, z = 5.0;
+    double dx = 0.0, dy = 0.0;
+    grad.execute(x, y, z, &dx, &dy);
+    printf("d_x = %.2f, d_y = %.2f\n", dx, dy);
+}
+```
+In this example,
+we have used `clad::differentiate` to generate a function `grad` that calculates
+the gradient of the function `prod` with respect to the variables `x and y` as we
+can see the references are passed. The significance of `clad::opts::vector_mode`
+is explained in the below sections.
+
+## Working
+
```

Comment:
In the start, before the code snippet, we can add some description to say something like - "This is how standard forward mode AD works", and then go on to explain the differences in vector mode with the help of the generated vectorized function.